mooncake-store/src/segment.cpp

```diff
@@ -76,8 +76,15 @@ ErrorCode ScopedSegmentAccess::MountSegment(const Segment& segment,
     }
 
     segment_manager_->allocators_.push_back(allocator);
-    segment_manager_->allocators_by_name_[segment.name].push_back(allocator);
     segment_manager_->client_segments_[client_id].push_back(segment.id);
+
+    if (segment.is_vram)
```

Comment:
You may want to modify `GetClientSegments` to also return vram segment. This function will be used in HA mode to auto unmount segments from clients that may have crashed. Also, you may want to check other segment-related interfaces in master_service for similar issues.